modules/ui/src/main/scala/lucuma/ui/components/LinkIfValid.scala

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.ui.components
+
+import cats.effect.IO
+import cats.effect.Temporal
+import cats.syntax.all.*
+import crystal.react.hooks.*
+import crystal.react.syntax.all.*
+import japgolly.scalajs.react.*
+import japgolly.scalajs.react.vdom.html_<^.*
+import lucuma.react.common.*
+import org.http4s.Method
+import org.http4s.Request
+import org.http4s.Uri
+import org.http4s.client.Client
+import org.typelevel.log4cats.Logger
+
+import scala.concurrent.duration.*
+import scala.concurrent.duration.FiniteDuration
+
+/**
+ * A component that renders a link if the link is valid, otherwise it just renders the children.
+ */
+case class LinkIfValid(href: String, mods: TagMod*)(val children: VdomNode*)(using
+  val client: Client[IO],
+  val logger: Logger[IO],
+  val F:      Temporal[IO]
+) extends ReactFnProps(LinkIfValid.component)
+
+object LinkIfValid:
+  private type Props = LinkIfValid
+
+  private val PollInterval: FiniteDuration = 5.seconds
+
+  private val ProxyUri: String = "https://cors-proxy.lucuma.xyz" // Avoid CORS
+
+  private val component =
+    ScalaFnComponent
+      .withHooks[Props]
+      .useState(false) // isValid
+      .useEffectStreamWithDepsBy((_, isValid) => isValid.value): (props, isValid) =>
+        isValudValue =>
+          import props.F
+
+          Option
+            .when(!isValudValue):
+              Uri
+                .fromString(s"$ProxyUri/${props.href}")
+                .bimap(
+                  parseError =>
+                    fs2.Stream.eval(props.logger.error(s"Error parsing URI: $parseError")),
+                  uri =>
+                    fs2.Stream
+                      .eval:
+                        props.logger.trace(s"Checking URI: $uri") >>
+                          props.client
+                            .successful(Request[IO](Method.HEAD, uri))
```

Comment:
Great idea, Do you think it will be a problem with hundreds off links?

Reply:
It shouldn't be a problem. Because of virtualization, this is only triggered for the datasets that are in screen. This would be a problem when scrolling, but I don't think it will be because browsers only allow a certain number of outgoing connections and queue the rest. When a link scrolls out of view, it is unmounted and canceled, so we should be OK.